src/pkg/clouds/aws/ecs/logs.go

```diff
@@ -62,56 +62,56 @@ func getLogGroupIdentifier(arnOrId string) string {
 	return strings.TrimSuffix(arnOrId, ":*")
 }
 
-func TailLogGroups(ctx context.Context, logGroups ...LogGroupInput) (EventStream, error) {
+func TailLogGroups(ctx context.Context, since time.Time, logGroups ...LogGroupInput) (EventStream, error) {
+	child, cancel := context.WithCancel(ctx)
 	var cs = collectionStream{
-		ch:    make(chan types.StartLiveTailResponseStream),
-		errCh: make(chan error),
-		done:  make(chan struct{}),
+		cancel: cancel,
+		ch:     make(chan types.StartLiveTailResponseStream, 10), // max number of loggroups to query
+		ctx:    child,
+		errCh:  make(chan error, 1),
```

Comment:
Nick Pick: I think it might be better to have all these init code in a `newCollectionStream` function